### DISCUSSION TITLE
Does one tonic::transport::Endpoint make use of multiple http2 connections?

### DISCUSSION THREAD

**Discussion by kingluo** [hyperium/tonic#1128] [Q&A] [answered]
Each client is associated with one endpoint, right? Does one endpoint create multiple http2 connections under the hood? And side question is, does it use a thread pool or tokio tasks pool to handle these connections?

**Comment by LucioFranco** [comment, marked as answer]
One Channel/Endpoint == one connection under the hood, EXCEPT if you use the balance feature which will then allow you to load balance many connections on a single Channel.

**Reply by kingluo** [reply]
Thanks for your reply.
Does balance feature aims at different dest addresses? one connection per address?
Is it possible to have multiple connections for the same address?
Another side question is, the connection has no reconnect mechanism, right?

**Reply by LucioFranco** [reply]
Balance can be between same or different endpoints. Yes, if a request fails because of a connection issue when you send the next request it will attempt to reconnect.
